typings/globals/missing.d.ts

```diff
@@ -1,74 +1,90 @@
-  matchAll(options: ServiceWorkerClientsMatchOptions): Promise<Array<WindowClient>>;
+interface Permissions {
+  query(permissionDesc: PermissionDescriptor | DevicePermissionDescriptor | MidiPermissionDescriptor | PushPermissionDescriptor): Promise<PermissionStatus>;
 }
 
-interface WindowClient extends ServiceWorkerClient {
-  /**
-   * Loads a specified URL into a controlled client page.
-   */
-  navigate(url: string): Promise<WindowClient>;
-}
+declare var Permissions: {
+  prototype: Permissions;
+  new(): Permissions;
+};
 
-interface PushMessageData {
-  arrayBuffer(): ArrayBuffer;
-  blob(): Blob;
-  json(): any;
-  text(): string;
+interface Navigator {
+  readonly permissions: Permissions;
 }
 
-interface PushEvent {
-  data?: PushMessageData
-}
 
-interface String {
-  endsWith(...args): any;
+interface Notification {
+  // Added in TypeScript 3.5.1
+  readonly permission: NotificationPermission;
 }
 
-interface ServiceWorkerGlobalScope {
-  addEventListener(type: "push", listener: (this: this, ev: any) => any, ...args): void;
-  addEventListener(type: "notificationclose", listener: (this: this, ev: any) => any, ...args): void;
-  addEventListener(type: "notificationclick", listener: (this: this, ev: any) => any, ...args): void;
-  addEventListener(type: "install", listener: (this: this, ev: any) => any, ...args): void;
-  addEventListener(type: "activate", listener: (this: this, ev: any) => any, ...args): void;
-  addEventListener(type: "pushsubscriptionchange", listener: (this: this, ev: any) => any, ...args): void;
-  addEventListener(type: "fetch", listener: (this: this, ev: any) => any, ...args): void;
-  readonly location: Location;
+/**
+ * END: Permission Definitions
+ */
+
+
+// FrameType - added in TypeScript 3.5.1
+//   https://github.com/microsoft/TypeScript/blob/v3.5.1/lib/lib.webworker.d.ts
+type FrameType = "auxiliary" | "top-level" | "nested" | "none";
+interface Client {
+  readonly frameType: FrameType;
 }
 
-interface TestContext {
-  context: any;
+/**
+ * START: window.safari definition
+ * https://developer.apple.com/documentation/safariextensions
+ */
+interface SafariRemoteNotificationPermission {
+  readonly deviceToken: string | null | undefined;
+  readonly permission: string;
 }
 
-declare module ExtendableError {
+interface SafariRemoteNotification {
+  permission(bundleIdentifier: string): SafariRemoteNotificationPermission;
+  requestPermission(
+    webAPIURL: string,
+    websiteIdentifier: string,
+    queryParameterDictionary: any,
+    callback: Function
+  ): void;
 }
 
-interface Element extends Node, GlobalEventHandlers, ElementTraversal, NodeSelector, ChildNode, ParentNode {
-  addEventListener<K extends keyof ElementEventMap>(type: K, listener: (this: Element, ev: ElementEventMap[K]) => any, optionsOruseCapture?: boolean | { passive: boolean }): void;
+interface Window {
+  Notification: any; // TODO: Remove as "Notification" should be used over "window.Notification"
```

Comment:
This TODO is being completed in https://github.com/OneSignal/OneSignal-Website-SDK/pull/566